packages/@aws-cdk/toolkit-lib/lib/api/cloud-assembly/private/prepare-source.ts

```diff
@@ -33,18 +49,39 @@ export class ExecutionEnvironment implements AsyncDisposable {
    * If `markSuccessful()` is called, the writer lock is converted to a reader lock
    * and temporary directories will not be cleaned up anymore.
    */
-  public static async create(services: ToolkitServices, props: { outdir?: string } = {}) {
-    let tempDir = false;
-    let dir = props.outdir;
+  public static async create(services: ToolkitServices, options: ExecutionEnvironmentOptions) {
+    let outDirIsTemporary = false;
+    let dir = options.outdir;
     if (!dir) {
-      tempDir = true;
+      outDirIsTemporary = true;
       dir = fs.mkdtempSync(path.join(fs.realpathSync(os.tmpdir()), 'cdk.out'));
```

Comment:
What happens here if there is already a `cdk.out` directory in the tmp dir location?